daemon/tests/happy_path.rs

```diff
@@ -86,28 +71,34 @@ async fn taker_takes_order_and_maker_rejects() {
     assert!(matches!(maker_cfd.state, CfdState::Rejected { .. }));
 }
 
+// Helper function setting up a "happy path" wallet mock
+fn mock_wallet_transaction(wallet: &mut MutexGuard<'_, harness::mocks::wallet::MockWallet>) {
+    #[allow(clippy::redundant_closure)] // clippy is in the wrong here
+    wallet
+        .expect_build_party_params()
+        .times(1)
+        .returning(|msg| build_party_params(msg));
+
+    wallet
+        .expect_sign()
+        .times(1)
+        .returning(|| Ok(dummy_partially_signed_transaction()));
+    wallet
+        .expect_broadcast()
+        .times(1)
+        .returning(|| Ok(dummy_tx_id()));
+}
+
 #[tokio::test]
 async fn taker_takes_order_and_maker_accepts_and_contract_setup() {
     let _guard = init_tracing();
-    let (mut maker, mut taker) = start_both(
-        Oracle::default()
-            .with_dummy_announcement(harness::cfd_protocol::OliviaData::example_0().announcement()),
-        Monitor,
-        Wallet::default()
-            .with_wallet_info()
-            .with_party_params()
-            .with_psbt(harness::bdk::dummy_partially_signed_transaction())
-            .with_txid(harness::bdk::dummy_tx_id()),
-        Oracle::default()
-            .with_dummy_announcement(harness::cfd_protocol::OliviaData::example_0().announcement()),
-        Monitor,
-        Wallet::default()
-            .with_wallet_info()
-            .with_party_params()
-            .with_psbt(harness::bdk::dummy_partially_signed_transaction())
-            .with_txid(harness::bdk::dummy_tx_id()),
-    )
-    .await;
+    let (mut maker, mut taker) = start_both().await;
+
+    mock_wallet_transaction(&mut maker.mocks.wallet().await);
+    mock_wallet_transaction(&mut taker.mocks.wallet().await);
+
+    maker.mocks.monitor().await.expect_sync().return_const(());
```

Comment:
I'm afraid this might cause brittle tests, because the sync depends on time intervals. If a test runs longer on e.g. CI we cannot guarantee we don't run into another sync interval; additionally, if we change the sync interval tests might break even though they should not be affected (we don't test the sync, we test other behavior...).

Would it be possible to just let all `sync` messages be ignored in the tests? i.e. is it possible to instruct the mocks to just not assert on `sync` messages? 
We can have specific tests that test sync sanity, but for other tests I don't think we should care about it. 
This also accounts for the oracle's sync.

Reply:
I'm having trouble understanding the brittleness - the way it stands, I just say to return `(())` whenever the `Sync` is called (note that there's no `times()` condition). It preserves the behaviour that we had before introduction of the mock 1:1.

the mocks don't assert anything in the tests so far, and will never count how many times `sync` has been called as we don't use the `times()` here.

Reply:
Oh, then I probably just misunderstood that - that is what we want then. Could still opt for configuring the monitor sync in the setup as well if we want.

Reply:
sure, I'll move it into the setup, so it doesn't get in the way.